[Enhancement] Supports runtime modification of some BE and Starlet params

stonechen111 opened the pull request:
## Why I'm doing:
Some BE and Starlet parameters cannot be modified in runtime and are inconvenient to use.

## What I'm doing:
The supported parameters are as follows:
Four BE parameters:
1. pipeline_connector_scan_thread_num_per_cpu
2. disable_column_pool
3. create_tablet_worker_count
4. number_tablet_writer_threads

Seven Starlet parameters:
1. starlet_cache_thread_num
2. starlet_cache_evict_low_water
3. starlet_cache_evict_high_water
4. starlet_fs_stream_buffer_size_bytes
5. starlet_fs_read_prefetch_enable
6. starlet_fs_read_prefetch_threadpool_size
7. starlet_cache_evict_interval

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

sduzh commented:
@wyb @kevincai PTAL

wyb commented on be/src/common/config.h:
```diff
@@ -268,7 +268,7 @@ CONF_mBool(enable_zonemap_index_memory_page_cache, "false");
 CONF_mBool(enable_ordinal_index_memory_page_cache, "false");
 // whether to disable column pool
-CONF_Bool(disable_column_pool, "true");
+CONF_mBool(disable_column_pool, "true");
```
if `disable_column_pool` is updated from `true` to `false`, we should clear the column pool?

stonechen111 replied:
> if `disable_column_pool` is updated from `true` to `false`, we should clear the column pool?

It does not take effect on the existing column pool, only on the new column pool.

kevincai commented on be/src/http/action/update_config_action.cpp:
```diff
@@ -171,6 +174,60 @@ Status UpdateConfigAction::update_config(const std::string& name, const std::str
+        });
+        _config_callback.emplace("create_tablet_worker_count", [&]() {
+            LOG(INFO) << "set create_tablet_worker_count:" << config::create_tablet_worker_count;
+            auto thread_pool = _exec_env->agent_server()->get_thread_pool(TTaskType::CREATE);
```
use `_exec_env` or `ExecEnv::GetInstance()` ?

kevincai commented on be/src/http/action/update_config_action.cpp:
```diff
@@ -171,6 +174,60 @@ Status UpdateConfigAction::update_config(const std::string& name, const std::str
+            auto thread_pool = _exec_env->agent_server()->get_thread_pool(TTaskType::CREATE);
+            (void)thread_pool->update_max_threads(config::create_tablet_worker_count);
+        });
+        _config_callback.emplace("number_tablet_writer_threads", [&]() {
```
it is so weird to register all config callback in an http action.

stonechen111 committed:
[Enhancement] Supports runtime modification of some BE and Starlet params

The supported parameters are as follows:
Three BE parameters:
1. pipeline_connector_scan_thread_num_per_cpu
2. create_tablet_worker_count
3. number_tablet_writer_threads

Seven Starlet parameters:
1. starlet_cache_thread_num
2. starlet_cache_evict_low_water
3. starlet_cache_evict_high_water
4. starlet_fs_stream_buffer_size_bytes
5. starlet_fs_read_prefetch_enable
6. starlet_fs_read_prefetch_threadpool_size
7. starlet_cache_evict_interval

Signed-off-by: stonechen111 <chenshi@starrocks.com>
